src/atoms/Switch/Switch.native.js

```diff
@@ -0,0 +1,315 @@
+import React, { useState, useContext, useRef, useEffect } from 'react';
+import { Animated, TouchableOpacity } from 'react-native';
+import styled, { ThemeContext } from 'styled-components/native';
+import PropTypes from 'prop-types';
+
+import { makePxValue } from '../../_helpers/theme';
+import automation from '../../_helpers/automation-attributes';
+import View from '../View/';
+import isPropDefined from '../../_helpers/isPropDefined';
+
+const styles = {
+  size({ size, theme }) {
+    switch (size) {
+      case 'medium':
+      default:
+        return {
+          containerSize: {
+            height: theme.spacings.large,
+            width: theme.spacings.xxxlarge,
+          },
+          knobSize: {
+            height: theme.spacings.medium,
+            width: theme.spacings.medium,
+            activeWidth: theme.spacings.large,
+          },
+        };
+      case 'large':
+        return {
+          containerSize: {
+            height: makePxValue(3),
+            width: makePxValue(6),
+          },
+          knobSize: {
+            height: theme.spacings.large,
+            width: theme.spacings.large,
+            activeWidth: theme.spacings.xxlarge,
+          },
+        };
+    }
+  },
+  padding({ size, theme }) {
+    switch (size) {
+      case 'medium':
+      default:
+        return theme.spacings.xxsmall;
+      case 'large':
+        return theme.spacings.xsmall;
+    }
+  },
+  alignSelf({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+};
+
+const StyledContainer = styled(TouchableOpacity)(
+  (props) => `
+  height: ${styles.size(props).containerSize.height};
+  width: ${styles.size(props).containerSize.width};
+  border-radius: ${`${parseInt(styles.size(props).containerSize.height, 10) / 2}px`};
+  padding: ${styles.padding(props)};
+  flex-direction: row;
+  align-self: ${styles.alignSelf(props)};
+`,
+);
```

Comment:
Do all of this calculation in ```styles``` object. This should look like this

```suggestion
const StyledContainer = styled(TouchableOpacity)`
  height: ${styles.size.containerSize.height};
  width: ${styles.size.containerSize.width};
  border-radius: ${styles.borderRadius};
  padding: ${styles.padding)};
  flex-direction: row;
  align-self: ${styles.alignSelf)};
`,
```